analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/BinaryBase.cs

```diff
@@ -0,0 +1,296 @@
+        ISymbol InstanceOfCountProperty(IOperation operation) =>
+            operation.AsPropertyReference() is { Instance: { } instance, Property.Name: nameof(Array.Length) or nameof(List<int>.Count) }
+            && instance.Type.DerivesOrImplementsAny(CollectionTracker.CollectionTypes)
+            && instance.TrackedSymbol(state) is { } symbol
+                ? symbol
+                : null;
+    }
+
+    private static SymbolicConstraint CollectionConstraintFromOperator(BinaryOperatorKind operatorKind, NumberConstraint number) =>
+        // consider count to be the left operand and the comparison to resolve to true:
+        operatorKind switch
+        {
+            _ when operatorKind.IsEquals() && number.Min > 0 => CollectionConstraint.NotEmpty,
+            _ when operatorKind.IsEquals() && number.Max == 0 => CollectionConstraint.Empty,
+            _ when operatorKind.IsNotEquals() && number.Max == 0 => CollectionConstraint.NotEmpty,
+            BinaryOperatorKind.GreaterThan when number.Min >= 0 => CollectionConstraint.NotEmpty,
+            BinaryOperatorKind.GreaterThanOrEqual when number.Min >= 1 => CollectionConstraint.NotEmpty,
+            BinaryOperatorKind.LessThan when number.Max == 1 => CollectionConstraint.Empty,
+            BinaryOperatorKind.LessThanOrEqual when number.Max == 0 => CollectionConstraint.Empty,
+            _ => null
+        };
+
+    private static BoolConstraint BoolConstraintFromBoolConstraints(BinaryOperatorKind kind, bool left, bool right) =>
+         kind switch
+         {
+             BinaryOperatorKind.Equals or BinaryOperatorKind.ObjectValueEquals => BoolConstraint.From(left == right),
+             BinaryOperatorKind.NotEquals or BinaryOperatorKind.ObjectValueNotEquals => BoolConstraint.From(left != right),
+             BinaryOperatorKind.And or BinaryOperatorKind.ConditionalAnd => BoolConstraint.From(left && right),
+             BinaryOperatorKind.Or or BinaryOperatorKind.ConditionalOr => BoolConstraint.From(left || right),
+             BinaryOperatorKind.ExclusiveOr => BoolConstraint.From(left ^ right),
+             _ => null
+         };
+
+    private static BoolConstraint BoolConstraintFromNullConstraints(BinaryOperatorKind kind, bool isNullLeft, bool isNullRight) =>
+        isNullLeft || isNullRight
+            ? kind switch
+            {
+                BinaryOperatorKind.Equals or BinaryOperatorKind.ObjectValueEquals => BoolConstraint.From(isNullLeft && isNullRight),
+                BinaryOperatorKind.NotEquals or BinaryOperatorKind.ObjectValueNotEquals => BoolConstraint.From(isNullLeft != isNullRight),
+                _ when kind.IsAnyRelational() => BoolConstraint.False,
+                _ => null
+            }
+            : null;
+
+    private static BoolConstraint BoolConstraintFromBoolAndNullConstraints(
+        BinaryOperatorKind kind,
+        BoolConstraint leftBool,
+        BoolConstraint rightBool,
+        bool leftIsNull,
+        bool rightIsNull) =>
```

Comment:
According to our coding style:
```suggestion
    private static BoolConstraint BoolConstraintFromBoolAndNullConstraints(BinaryOperatorKind kind,
                                                                           BoolConstraint leftBool,
                                                                           BoolConstraint rightBool,
                                                                           bool leftIsNull,
                                                                           bool rightIsNull) =>
```